http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClient.java

```diff
@@ -1070,27 +1177,32 @@ public Publisher<MutableHttpResponse<?>> proxy(@NonNull io.micronaut.http.HttpRe
     public Publisher<MutableHttpResponse<?>> proxy(@NonNull io.micronaut.http.HttpRequest<?> request, @NonNull ProxyRequestOptions options) {
         Objects.requireNonNull(options, "options");
         setupConversionService(request);
-        return Flux.from(resolveRequestURI(request))
+        return resolveRequestURI(request)
                 .flatMap(requestURI -> {
                     MutableHttpRequest<?> httpRequest = toMutableRequest(request);
                     if (!options.isRetainHostHeader()) {
                         httpRequest.headers(headers -> headers.remove(HttpHeaderNames.HOST));
                     }
 
-                    AtomicReference<MutableHttpRequest<?>> requestWrapper = new AtomicReference<>(httpRequest);
-                    Flux<HttpResponse<?>> proxyResponsePublisher = connectAndStream(request, request, requestURI, requestWrapper, true, false);
-                    // apply filters
-                    //noinspection
-                    proxyResponsePublisher = Flux.from(
-                            applyFilterToResponsePublisher(
-                                    request,
-                                    requestWrapper.get(),
-                                    requestURI,
-                                    proxyResponsePublisher
-                            )
+                    return this.sendRequestWithRedirects(
+                        request,
+                        null,
+                        httpRequest.uri(requestURI),
+                        (req, resp) -> {
```

Comment:
Please extract a new method

Reply:
imo extracting these lambdas makes the code more spaghetti because it's not immediately clear that this code is only called in one place.